File: string/1957_delete_characters_to_make_fancy_string/1957_delete_characters_to_make_fancy_string.go
package main

func makeFancyString(s string) string {
	res := []string{}
	for _, c := range s {
		if len(res) >= 2 && res[len(res)-1] == string(c) && res[len(res)-2] == string(c) {
			continue
		}
		res = append(res, string(c))
	}
	return strings.Join(res, "")
}

File: string/1957_delete_characters_to_make_fancy_string/1957_delete_characters_to_make_fancy_string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        res = []
        for i in s:
            if len(res) >= 2 and i == res[-1] and i == res[-2]:
                continue
            else:
                res.append(i)
        return ''.join(res)
